src/header/common/cookie.rs

```diff
@@ -154,6 +159,17 @@ impl fmt::Display for Cookie {
     }
 }
 
+#[allow(missing_debug_implementations)]
```

Comment:
If I were to take an educated guess, I would say the lint didn't trigger because `CookieIter` is not exported all the way to the crate API.

@mayah, you might want to add a `pub use cookie::CookieIter` to [src/header/common/mod.rs](https://github.com/hyperium/hyper/blob/master/src/header/common/mod.rs#L33) to make sure `CookieIter` is exported from the crate.